### PR TITLE
Build fewer tests in Travis platform_dependent tests

### PR DESCRIPTION
**Pull request by siying**
Summary: platform_dependent tests in Travis now builds all tests, which is not needed. Only build those tests we need to run.

### CODE REVIEW

File: db/db_basic_test.cc [hunk 1]
```diff
@@ -175,6 +175,8 @@ TEST_F(DBBasicTest, LevelLimitReopen) {
   int i = 0;
   while (NumTableFilesAtLevel(2, 1) == 0) {
     ASSERT_OK(Put(1, Key(i++), value));
+    dbfull()->TEST_WaitForFlushMemTable();
+    dbfull()->TEST_WaitForCompact();
```
Comment (IslamAbdelRahman) [inline db/db_basic_test.cc]: Looks like this is not related to this PR

**Comment by IslamAbdelRahman** [comment]
Is this failure related ?
https://travis-ci.org/facebook/rocksdb/jobs/257845757

I tried to restart the test and it still appeared

**Comment by siying** [comment]
@IslamAbdelRahman I've no idea. It doesn't reproduce on my dev server.

File: db/db_basic_test.cc (hunk 1 above)
Comment (siying) [inline db/db_basic_test.cc]: Yes, but I think this is better because two of the previous out-of-space failures were in this test.

**Comment by siying** [comment]
Fixed a bug. But I don't think it is related to the failure:(

**Comment by IslamAbdelRahman** [comment]
looks like the new commit don't have the changes to the Makefile
